gmso/core/topology.py

```diff
@@ -647,6 +659,21 @@ def get_scaling_factors(self, *, molecule_id=None):
             ]
         )
 
+    def remove_site(self, site):
```

Comment:
I think remove site should automatically iterate through the connections and remove them. That loop that's in the gist should be part of remove_site.
```suggestion
    def remove_site(self, site):
        site_connDict = self.get_connections_by_site(site)
        for conn_type in site_connDict:
            for conn in site_connDict[conn_type]:
                self.remove_connection(conn)
       self._sites.remove(site)
```